### ISSUE TITLE
generate-dump no longer calls hw-management-generate-dump.sh

### ISSUE THREAD

**Issue by Staphylo** [sonic-net/sonic-buildimage#16524]
#### Description

The techsupport command no longer runs `/usr/bin/hw-management-generate-dump.sh` for non-mellanox products.
This issue originate from the refactor of `generate_dump` from following PR https://github.com/sonic-net/sonic-utilities/pull/2117
I left a message on the PR but since I haven't gotten any response I'm opening a proper issue here.

This `hw-dump` mechanism was really useful for platform vendors to add their own hooks and complement the generic dump with platform specific information.
This results in a loss of important debugging information during support cases.

The goal of this issue is to understand if we could move this mechanism back to being part of the generic code of `generate_dump`.
If this is not desirable by the Nvidia team, there would be a need to come up with an alternative that does pretty much the same thing.

#### Steps to reproduce the issue:
1. run `generate_dump` on any arista product, notice `hw-dump` folder missing.

#### Output of `show version`:

This happens any release branch with the code change mentioned originally.
Therefore 202205+

**Comment by liat-grozovik** [comment]
@stepanblyschak  can you please check the above?
in any case the platform dump should be call and the hw-mgmt dump is nvidia specific.

**Comment by stepanblyschak** [comment]
@Staphylo hw-management-generate-dump.sh is Nvidia platform specific script that comes with hw management package (https://github.com/Mellanox/hw-mgmt).
It wasn't intendend to be used as a way for other vendors to collect dump. The fact it could be used so was a bug. As you can see in the script there are platform specific routines - ```collect_mellanox```, ```collect_broadcom```. So, you can add youre platform specific routine to the script to fix it or implement a plugin (which is part of App. Extension infrastructure to extend techsupport dump) - check ```PLUGINS_DIR```

**Comment by Staphylo** [comment]
Thank you both for your answers.

We ended up implementing `hw-management-generate-dump.sh` to get some information included in the dump but missed the introduction of the `PLUGINS_DIR` hooks option.
I completely understand that `hw-mgmt` is a nvidia specific mechanism and should be retained as such.
I will therefore update our platform package to go down the `PLUGINS_DIR` way.

Closing this issue as solved.
